src/coreclr/tools/Common/CommandLineHelpers.cs

```diff
@@ -69,22 +69,21 @@ public static TargetOS GetTargetOS(string token)
 
                 throw new NotImplementedException();
             }
-            else
+
+            string lowerToken = token.ToLowerInvariant();
+            return lowerToken switch
             {
-                return token.ToLowerInvariant() switch
-                {
-                    "windows" => TargetOS.Windows,
-                    "linux" => TargetOS.Linux,
-                    "freebsd" => TargetOS.FreeBSD,
-                    "osx" => TargetOS.OSX,
-                    "maccatalyst" => TargetOS.MacCatalyst,
-                    "ios" => TargetOS.iOS,
-                    "iossimulator" => TargetOS.iOSSimulator,
-                    "tvos" => TargetOS.tvOS,
-                    "tvossimulator" => TargetOS.tvOSSimulator,
-                    _ => throw new CommandLineException($"Target OS '{token}' is not supported")
-                };
-            }
+                "linux" or "linux-musl" => TargetOS.Linux,
+                _ when lowerToken.StartsWith("win") => TargetOS.Windows,
```

Comment:
It would be better if accepted exact matches only. Does anything depend on the fuzziness of `StartsWith` matches here?

Reply:
In the current revision, targets shouldn't pass `win10` etc. (which was needed in earlier revision when we were passing values without using StartsWith in the targets).